### DISCUSSION TITLE
How to set the Porosity and of aquifers in flopy

### DISCUSSION THREAD

**Discussion by Alvin-prc** [modflowpy/flopy#1764] [Q&A] [answered]
I can use the LPF packet to set the  Specific Yield, but I can't find a parameter to set the effective porosity

**Comment by emorway-usgs** [comment]
Specific yield is necessary for the storage calculations in the flow model.  Porosity is needed for calculating velocities in a transport model.  You will find porosity in the BTN package of MT3D, or you will find it in the MST package of MODFLOW 6.

**Comment by ccalvocm** [comment, marked as answer]
Greetings @Alvin-prc 

Porosity is not directly used in flow packages because the transient groundwater flow equation is expressed in terms of Specific yield and specific storage rather than aquifer compressibility ($\alpha$), water compressibility ($\beta$) and porosity ($\eta$), so Groundwater flow equation:

$\nabla \cdot (K \nabla h)+W=(\alpha \rho_w g+\eta \beta \rho_w g) \frac{\partial h}{\partial t}$ 

Where W stands for a source term, $\rho_w$ is water density. Rewriting the GWF equation in terms of storativity becomes:

$\nabla \cdot (K \nabla h)+W=S \frac{\partial h}{\partial t}$

Where $S=S_y+b S_s$ stands for storativity, being b the confined aquifer thickness, while specific storage ($S_s$ ) and specific yield ($S_y$)  are specified for MODFLOW transient simulation, porosity is inherently present in those terms but not directly used. 

On the other hand, porosity $\eta$ is used in transport packages as @emorway-usgs stated, and also in MODPATH, where the specific linear velocity  ($v=\frac{Q}{\eta A}$) is used. Porosity is specified in the MODPATH basic data file, flopy.modpath.mp7bas on flopy: https://flopy.readthedocs.io/en/3.3.6/_modules/flopy/modpath/mp7bas.html

References:

- Harbaugh, A. W. (2005). MODFLOW-2005, the US Geological Survey modular ground-water model: the ground-water flow process (Vol. 6). Reston, VA, USA: US Department of the Interior, US Geological Survey.
- Niswonger, R. G., Panday, S., & Ibaraki, M. (2011). MODFLOW-NWT, a Newton formulation for MODFLOW-2005. US Geological Survey Techniques and Methods, 6(A37), 44.
- Pollock, D. W. (2012). User guide for MODPATH version 6: a particle tracking model for MODFLOW (p. 58). Reston, VA, USA: US Department of the Interior, US Geological Survey.

**Reply by Alvin-prc** [reply]
Thank you
